CyberwatchApi.psm1

```diff
@@ -64,74 +80,108 @@ Class CbwApiClient {
         return SendApiRequest -api_url $this.api_url -api_key $this.api_key -secret_key $this.secret_key -http_method $http_method -request_URI $request_URI -content $content
     }
 
+    [object] request_json([string]$http_method, [string]$request_URI) {
+        return $this.request($http_method, $request_URI) | ConvertFrom-JSON
+    }
+
+    [object] request_json([string]$http_method, [string]$request_URI, [Hashtable]$content) {
+
+        return $this.request($http_method, $request_URI, $content) | ConvertFrom-JSON
+    }
+
+    [object] get_all_pages([string]$http_method,[string]$request_URI, [Hashtable]$filter) {
+
+        $response =  $response = $this.request($http_method, $request_URI, $filter)
+        if ($response.headers["link"] -match "[?&]page=(\d*)") {
+                $last_page_number = $matches[1]
+            }
+         else { throw "Error: last page number not found" }
+
+        $result = ForEach ($i in 1..$last_page_number) {
```

Comment:
Mon commentaire demande si on peut ou non stream le resultat. Je cherche vraiment pas à t'embéter sur la syntaxe du for ou foreach ou ce que tu veux, je cherche juste a ce que l'utilisateurs est les résultats au fur et mesure non pas en un block après avoir attendu deux heures.